proxy/proxy.ts

```diff
@@ -13,17 +13,179 @@ class BadRequestError extends Error {
   }
 }
 
-export function createProxyServer(config: {
+interface RateLimitInfo {
+  count: number
+  timestamp: number
+}
+
+interface ProxyConfig {
   allowLocalhost?: boolean
   allowsFrom: RegExp[]
   maxSize: number
   timeout: number
-}): Server {
-  return createServer(async (req, res) => {
+}
+
+const RATE_LIMIT = {
+  PER_DOMAIN: {
+    LIMIT: 500,
+    DURATION: 60 * 1000
+  },
+  GLOBAL: {
+    LIMIT: 5000,
+    DURATION: 60 * 1000
+  }
+}
+
+const delay = (ms: number): Promise<void> => {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+const rateLimitMap: Map<string, RateLimitInfo> = new Map()
```

Comment:
All variables which is not `NAMED_LIKE_THIS` should be `let` according to our code style

Reply:
> All variables which is not `NAMED_LIKE_THIS` should be `let` according to our code style

So I should not acknowledge constant references and use `const` only for real constants?

Reply:
Yes, use `let` for variables (ESLint will tell you some cases when you run it)